### PR TITLE
Add V1 token migration transaction hooks

### PR DESCRIPTION
**Pull request by aeolianeth**
## What does this PR do and why?

More prep for https://github.com/jbx-protocol/juice-interface/issues/1335 (https://github.com/jbx-protocol/juice-interface/issues/1056)

⚠️ There are no user-facing changes in the PR

Adds transaction hooks:
- PayV1TokenPaymentTerminal
- V1SetOperator

In addition:
- minor V2ContractLoader refactor to include new terminal contract
- move and rename some files
- update contract reader logging
- clean up v2 manage tokens section pre-render

## Screenshots or screen recordings

No user-facing changes.

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/v2/V2Project/V2ManageTokensSection/index.tsx [hunk 1]
```diff
@@ -29,10 +29,18 @@ import V2RedeemModal from './V2RedeemModal'
 import V2ClaimTokensModal from './V2ClaimTokensModal'
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2ManageTokensSection/index.tsx:29]: This file's diff is pure housekeeping.

No user-facing or functional changes.

File: src/hooks/v1/transactor/V1SetOperatorTx.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import { TransactorInstance } from '../../Transactor'
+
+type OperatorData = {
+  operator: string
+  domain: number // projectId (or 0, for any project)
+  permissionIndexes: number[]
+}
+
+export function useV1SetOperatorTx(): TransactorInstance<OperatorData> {
```
Comment (aeolianeth) [inline src/hooks/v1/transactor/V1SetOperatorTx.ts:13]: We'll use this in the UI for https://github.com/jbx-protocol/juice-interface/issues/1335

File: src/hooks/v2/V2ContractLoader.ts [hunk 3]
```diff
@@ -73,6 +77,58 @@ export function useV2ContractLoader() {
...
+) => {
+  const contractOverride = CONTRACT_ABI_OVERRIDES[contractName]
+  const version = contractOverride?.version ?? 'latest'
+  const filename = contractOverride?.filename ?? contractName
+  return await import(
+    `@jbx-protocol/contracts-v2-${version}/deployments/${network}/${filename}.json`
+  )
+}
+
+const loadJBV1TokenPaymentTerminalContract = async (network: NetworkName) => {
```
Comment (aeolianeth) [inline src/hooks/v2/V2ContractLoader.ts:115]: Custom loader for the new `JBV1TokenPaymentTerminal` contract.

File: src/hooks/v2/V2ContractLoader.ts [hunk 4]
```diff
@@ -73,6 +77,58 @@ export function useV2ContractLoader() {
   return contracts
 }
 
+const loadJBProjectHandlesContract = async (network: NetworkName) => {
```
Comment (aeolianeth) [inline src/hooks/v2/V2ContractLoader.ts:80]: This is the same as previous. Just a minor quality of life refactor.

File: src/hooks/v2/contractReader/V1ProjectIdOfV2Project.ts [hunk 5]
```diff
@@ -1,42 +1,14 @@
...
-
-import { readProvider } from 'constants/readProvider'
-import { readNetwork } from 'constants/networks'
-import { ContractReadResult } from './V2ContractReader'
+import useV2ContractReader, { ContractReadResult } from './V2ContractReader'
 
 export function useV1ProjectIdOfV2Project(
   projectId: BigNumberish | undefined,
 ): ContractReadResult<BigNumber | undefined> {
-  const [loading, setLoading] = useState<boolean>(false)
```
Comment (aeolianeth) [inline src/hooks/v2/contractReader/V1ProjectIdOfV2Project.ts:15]: Now that we're loading the `JBV1TokenPaymentTerminal` contract in the V2 contract loader, we can use it here :)

File: src/hooks/v2/transactor/PayV1TokenPaymentTerminal.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+const DEFAULT_MIN_RETURNED_TOKENS = 0
+
+type PayV2ProjectTx = TransactorInstance<{
+  memo: string
+  preferClaimedTokens: boolean
+  beneficiary?: string
+  value: BigNumber
+}>
+
+export function usePayV1TokenPaymentTerminal(): PayV2ProjectTx {
```
Comment (aeolianeth) [inline src/hooks/v2/transactor/PayV1TokenPaymentTerminal.ts:20]: New Tx, used in https://github.com/jbx-protocol/juice-interface/issues/1335

File: src/hooks/v2/transactor/SetV1ProjectIdTx.ts [hunk 7]
```diff
@@ -1,38 +1,30 @@
...
 export function useSetV1ProjectIdTx(): TransactorInstance<{
   v1ProjectId: number
 }> {
-  const { transactor } = useContext(V2UserContext)
+  const { transactor, contracts } = useContext(V2UserContext)
   const { projectId } = useContext(V2ProjectContext)
   const { signingProvider } = useContext(NetworkContext)
 
   return ({ v1ProjectId }, txOpts) => {
-    const contractAddress =
```
Comment (aeolianeth) [inline src/hooks/v2/transactor/SetV1ProjectIdTx.ts:20]: Same as https://github.com/jbx-protocol/juice-interface/pull/1398/files#r919648301

File: src/models/v2/contracts.ts [hunk 8]
```diff
@@ -12,8 +12,12 @@ export enum V2ContractName {
   JBTokenStore = 'JBTokenStore',
   JBSingleTokenPaymentTerminalStore = 'JBSingleTokenPaymentTerminalStore',
   JBETHERC20ProjectPayerDeployer = 'JBETHERC20ProjectPayerDeployer',
+
   JBProjectHandles = 'JBProjectHandles',
   PublicResolver = 'PublicResolver',
+
+  JBV1TokenPaymentTerminal = 'JBV1TokenPaymentTerminal',
```
Comment (aeolianeth) [inline src/models/v2/contracts.ts:19]: New

**Comment by aeolianeth** [comment]
@wraeth-eth hey there, could you please review this when you get a chance? 😃

**Comment by aeolianeth** [comment]
@wraeth-eth scratch that - going to send @johnnyd-eth 's way as he needs this context for NFT rewards.

**Comment by johnnyd-eth** [comment]
Great shit, love what you've done with the contract loader. Will take note for the NFT contract 🧠 LGTM

**Review by johnnyd-eth — APPROVED** [review]

**Pull request merged**